juju/juju
# Remove novote attribute from machine

## Description of change

A previous PR introduced a new controller node which has the "wants vote" and "has vote" attributes for a controller.
The PR removes the machine NoVote attribute and modifies the peer grouper and enable ha code to use the new controller node attribute.
It also deletes old unused ha code which supported promoting and demoting machines as a controller; we no longer do that and instead remove machines no longer needed as a controller.

There's still a bit of cruft until the "hasvote" attribute is removed.

## QA steps

bootstrap
enable-ha
remove a controller, check ha status
enable ha again, check status
remove mongo primary, check status
add a machine
enable-ha --to, check status
